tests/func-tests/node_placement_test.go

```diff
@@ -0,0 +1,162 @@
+package tests_test
+
+import (
+	"encoding/json"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	"time"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	testscore "kubevirt.io/kubevirt/tests"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/apis"
+	networkaddonsv1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests/flags"
+
+	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+)
+
+var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", func() {
+
+	var workloadsNode *v1.Node
+
+	tests.FlagParse()
+	client, err := kubecli.GetKubevirtClient()
+	testscore.PanicOnError(err)
+
+	workloadsNodes, err := client.CoreV1().Nodes().List(k8smetav1.ListOptions{
+		LabelSelector: "node.kubernetes.io/hco-test-node-type==workloads",
+	})
+	testscore.PanicOnError(err)
+
+	if workloadsNodes != nil && len(workloadsNodes.Items) == 1 {
+		workloadsNode = &workloadsNodes.Items[0]
+		fmt.Fprintf(GinkgoWriter, "Found Workloads Node. Node name: %s; node labels:\n", workloadsNode.Name)
+		w := json.NewEncoder(GinkgoWriter)
+		w.SetIndent("", "  ")
+		w.Encode(workloadsNode.Labels)
+	}
+
+	BeforeEach(func() {
+		if workloadsNode == nil {
+			Skip("Skipping Node Placement tests")
+		}
+		tests.BeforeEach()
+	})
+
+	Context("validate node placement in workloads nodes", func() {
```

Comment:
General comment about this `Describe`:

The tests here are pretty large. In order to make them more readable, please consider to split them to some helper methods.

Please consider to wrap the `Expeced`s with `By` with a meaningful text, for clearity.

Reply:
done